opendevin/server/session.py

```diff
@@ -101,10 +131,20 @@ async def start_task(self, start_event):
             return
         self.agent_task = asyncio.create_task(self.controller.start_loop(task), name="agent loop")
 
-    def on_agent_event(self, event):
-        evt = {
-            "action": event.action,
-            "message": event.get_message(),
-            "args": event.args,
-        }
+    def on_agent_event(self, event: Observation | Action):
+
+        # TODO: trying to find a potentially better solution here
+        # but not sure what "message" is for
+        if isinstance(event, Observation):
+            evt = {
+                "action": event.__class__.__name__,
+                "message": event.content,
+                "args": event.__dict__,
+            }
+        else:
+            evt = {
+                "action": event.__class__.__name__,
+                "args": event.__dict__,
+                "message": event.message,
+            }
```

Comment:
@rbren Fix most issues related to the new backend server and should be ready for review!
But I could need help here -- what would the message generally contain? Do we have a list of different events/actions we supported with the front-end? Thanks :)

Reply:
I'm hoping we can use the `Action.to_dict` method to make this cleaner. We'll need a robust way to serialize/deserialize actions and observations so we can send them back and forth.

Right now, every payload has three fields:
* `action`: the ID of the action (could be something passive, like `output`)
* `args`: any extra data about the action. key-value pairs, but something we'll want to standardize. E.g. the `run` action has a `command` arg, and the `output` action has an `output` arg.
* `message`: a human-readable description of this action, to be put in the chat window


But it might make sense to split Actions and Observations with two distinct payloads. Maybe something like:
```json
{
  "action": "run",
  "args": {"command": "ls"},
  "message": "let's check what files are available"
}
```
and
```json
{
  "observation": "run",
  "content": "foo.txt\nbar.txt",
  "extras": {
    "exit_code": 0
  },
  "message": "I see two text files."
}
```

Reply:
Great suggestion! I totally agree! I probably need to read more about how front-end use these event - I can update the PR to do something like this:
```json
{
"type": "observation" | "action",
"name": "name_of_observation_or_action",
"arguments": {"kwargs for actions"},
"content": "TEXT CONTENT FOR OBS, null for action",
"message": 
"Human readable description of the action - not necessarily need to be model generated",
"extras": {
EXTRA STUFF
}
}
```